Assets/Scripts/Manager/GameManager.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using UnityEngine;
+
+public class GameManager : MonoBehaviour
+{
+    private static GameManager _instance;
+    private GameState _currentState;
+    public GameState CurrentState => _currentState;
+    public event Action<GameState> OnGameStateChanged;
+
+    // Input
+    private PlayerInput_Actions _playerInputActions;
+    public PlayerInput_Actions InputActions => _playerInputActions;
+
+    public static GameManager Instance {
+        get {
+            if (_instance == null) 
+                Debug.LogError("GameManager is null!");
+            return _instance;
+        }
+    }
+
+    void Awake() {
+        if (_instance != null && _instance != this) {
+            Destroy(this);
+            return;
+        }
+        
+        _instance = this;
+        DontDestroyOnLoad(gameObject);
+
+        setPlayerInputActions();
+        
+    }
+
+    private void setPlayerInputActions() {
```

Comment:
alle funktionen groß hattest du gesagt
```suggestion
    private void SetPlayerInputActions() {
```